api/v1beta1/machinehealthcheck_webhook.go

```diff
@@ -165,8 +176,12 @@ func (m *MachineHealthCheck) validate(old *MachineHealthCheck) error {
 		)
 	}
 
-	if len(allErrs) == 0 {
-		return nil
+	if len(m.Spec.UnhealthyConditions) == 0 {
+		allErrs = append(allErrs, field.Forbidden(
```

Comment:
Should we get rid of the kubebuilder marker? Just would want to avoid duplication in general

Reply:
I don't think getting rid of the marker is great as external consumers could be using this as included validation of those fields, right? I think we talked about getting rid of these before and decided against it.

Agreed on duplication though, and this is definitely something that could be done in a future api version, just not sure if it's possibly breaking with no benefit right now.

Reply:
Let's keep the markers in the current PR to avoid discussions and delay.

I'll open an issue for it and try to get the framing right :)

Reply:
(just resolve)

Reply:
Created issue: https://github.com/kubernetes-sigs/cluster-api/issues/7634